pkg/apis/nmstate/v1alpha1/condition_types.go

```diff
@@ -0,0 +1,21 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:openapi-gen=true
+type ConditionType string
+
+// +k8s:openapi-gen=true
+type Condition struct {
+	Type               ConditionType          `json:"type"`
+	Status             corev1.ConditionStatus `json:"status"`
+	Reason             string                 `json:"reason,omitempty"`
+	Message            string                 `json:"message,omitempty"`
+	LastHeartbeatTime  metav1.Time            `json:"lastHearbeatTime,omitempty"`
+	LastTransitionTime metav1.Time            `json:"lastTransitionTime,omitempty"`
+}
+
+type ConditionList []Condition
```

Comment:
Because ConditionList is a beast on its own, it has its own semantics. It never has two conditions of the same Type, it is a treated as a whole. That's why I think it makes sense to explicitly set it as a type.

Reply:
Can we have the other functions to be methods then?

Reply:
We can't, we cannot edit the list in-place, we need to generate a new list and reassign it to the parent structure.